[controllers-films-users] Add controllers

PascalABCD opened the pull request:
- UserController
- FilmController
- Exception an ReleaseDateValidator
- Basic happy path Tests

pascalplata committed:
[controllers-films-users] Add:

- UserController
- FilmController
- Exception an ReleaseDateValidator
- Basic happy path Tests

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,57 @@
+        user.setId(id++);
+
+        // если нет имени, используем логин
+        if (user.isNameBlank()) {
```
эта логика нужна и при обновлении пользователя, поэтому вынеси в отдельный внутренний метод

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,30 @@
+    private Long id;
+
+    @Email(message = "Email не корректен")
+    private String email;
```
еще нужна аннотация NotNull  с комментарием

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,30 @@
+
+    @Past(message = "Дата рождения не может быть в будущем")
+    @JsonFormat(pattern = "yyyy-MM-dd")
+    private LocalDate birthday;
```
и сюда тоже

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/validator/ReleaseDateValidator.java:
```diff
@@ -0,0 +1,16 @@
+
+    @Override
+    public boolean isValid(LocalDate date, ConstraintValidatorContext context) {
+        return date == null || !date.isBefore(MIN_RELEASE_DATE);
```
1) атрибут releaseDate неявно через требование быть позже MIN_RELEASE_DATE является обязательным 
2) чтобы убрать плохо читаемую инверсию "!", можно метод isBefore() заменить на isAfter()

PascalABCD replied:
@irek-valiev я не понимаю, что значит первый пункт :-(

irek-valiev replied:
Это значит, что условие валидности для releaseDate, что дата не null И позже MIN_RELEASE_DATE

pascalplata committed:
[controllers-films-users] review fixes:

- fix ReleaseDateValidator conditions
- add setDefaultName() instead of repeated isBlank checks
- set @NotNull annotations for email and birthday

pascalplata committed:
[controllers-films-users] remove unnecessary dependency

pascalplata committed:
[controllers-films-users] update
